File: Java/String-2/src/CatDog.java
public class CatDog {
    public static void main(String[] args) {
        /*
         * Return true if the string "cat" and "dog" appear the same number of times in the given string.
         *
         * catDog("catdog") → true
         * catDog("catcat") → false
         * catDog("1cat1cadodog") → true
         */
        System.out.println(catDog("catdog"));
        System.out.println(catDog("catcat"));
        System.out.println(catDog("1cat1cadodog"));
    }

    public static boolean catDog(String str) {
        int countCat = 0;
        int countDog = 0;
        for (int i = 0; i < str.length() - 2; i++) {
            if (str.startsWith("cat", i)) countCat++;
            if (str.startsWith("dog", i)) countDog++;
        }
        return countCat == countDog;
    }
}

File: Python/String-2/cat_dog.py
def cat_dog(str):
    """
    Return True if the string "cat" and "dog" appear the same number of times in the given string.

    cat_dog('catdog') → True
    cat_dog('catcat') → False
    cat_dog('1cat1cadodog') → True
    """
    cat = 0
    dog = 0
    for i in range(len(str) - 2):
        if str[i:i + 3] == "cat":
            cat += 1
        if str[i:i + 3] == "dog":
            dog += 1
    return cat == dog


print(cat_dog('catdog'))
print(cat_dog('catcat'))
print(cat_dog('1cat1cadodog'))
